### ISSUE TITLE
Phobos Episode 1 error on launch

### ISSUE THREAD

**Issue by hasnogames** [dhewm/dhewm3#213]
ERROR: Error: file script/phobos_events.script, line 6: Unknown event 'GetSelfEntity'

Link to the mod https://www.moddb.com/mods/phobos/downloads/phobos-episode-1-v10

**Comment by DanielGibson** [comment]
Yeah, for this to work the mods game dll sourcecode must be released under GPL and be ported to dhewm3 (see https://github.com/dhewm/dhewm3-sdk - I could do the porting if they give me the source).

I'm not in contact with the Phobos developers (yet?), but it could sure help if you (and everyone else who'd like to play it on dhewm3) told them about it so they know that people want this :-)

**Comment by Kr3m** [comment]
Forgive my ignorance, but whatever happened to the equivalent of QVM files that were used with idTech 3 so .so and .dll files were unnecessary for cross platform mod support?

**Comment by RalfVB** [comment]
Any and all requests to publish the source code have fallen on deaf ears thus far. They copy-pasted a response to anyone asking for it.

**Comment by DanielGibson** [comment]
Well, I wouldn't say "on deaf ears", from what I've seen on moddb they first said they'll look into it "when all episodes are done", and more recently they said "once the next episode is done".
I mean it would sure be great if they did it now, but at least it doesn't sound like it'll never happen.

@Kr3m: Well, Q1 and Q3A had QuakeC/QVM, Quake2 had native DLLs (=> .so on Linux, .dylib on Mac) and so does Doom3.. though Doom3 also has Doomscript which is used in addition to the DLLs which I think is a bit similar to QuakeC?
Not sure why they did it that way, but possible factors are:
* DLLs have better performance than interpreted bytecode (and the Doom3 game DLLs contain lots of physics code which needs good performance)
* Doom3 was written in C++, while Quake3 was written in C, and writing a QVM-like compiler for C++ would have been a lot harder (especially considering clang+LLVM didn't exist yet)

**Comment by RalfVB** [comment]
That's the answer they pasted as a reply to anyone asking for the source.
'Maybe when all episodes are done' can be safely interpreted as 'we cannot be bothered, ask again in over a decade'.

**Comment by SimonNJensen** [comment]
Phobos dev here. We are a small team working on Phobos and we can't work on everything at once. We are prioritizing Episode 2 over dhewm3 support atm.

Plus, we are still making tons of changes to the SDK. We will probably take a look at it after Episode 2 has beeen released. Which is not going to take another decade :)

**Comment by DanielGibson** [comment]
Hi Simon, that sounds great!
I'm glad to help you with porting once you're ready :)

**Comment by jjbarr** [comment]
@SimonNJensen Since E2 has been released, is there any possibility that the porting process might start?

**Comment by SimonNJensen** [comment]
Episode 2 took a lot longer than we anticipated. We're a small team and we're very focused on getting this game finished as soon as possible. So all our spare time will go towards the next episode before anything else.

**Comment by DanielGibson** [comment]
If you release the source I can do the porting (when I have some time myself, which unfortunately is not right now, but I should have more time later this year) :)

**Comment by Sol1vaN** [comment]
> Yeah, for this to work the mods game dll sourcecode must be released under GPL and be ported to dhewm3 (see https://github.com/dhewm/dhewm3-sdk - I could do the porting if they give me the source).
> 
> I'm not in contact with the Phobos developers (yet?), but it could sure help if you (and everyone else who'd like to play it on dhewm3) told them about it so they know that people want this :-)

well, I want to play this with your engine, so, it's possible that you contact the Phobos devs?

**Comment by SimonNJensen** [comment]
Phobos dev here. We will probably look into this after all episodes have been released. We don't have time atm.

**Comment by j4reporting** [comment]
I understand you don't want to release the source code at this time. Would it be feasible to give @DanielGibson access to the source code so that, time permitting, he could port the first two episodes to dhewm3  and release just the ported binaries? The source port runs much better than the vanilla engine.

**Comment by jjbarr** [comment]
I believe legally speaking that's a little hairy.

The dhewm3 SDK is under GPL. As none of us have the actual authority to have put it there, that implies that it's built on code that was GPLed by id. Unless there's some kind of dual licensing situation (AFAICT there isn't), that means that anyone who uses to Dhewm3 SDK to port their game up to Dhewm3 is *legally compelled* to release the source code to their game dll.

TL;DR, we're not getting a D3P port until the devs are comfortable releasing the source code.

**Comment by j4reporting** [comment]
yes, you may be right.  the original SDK was not under GPL but the dhewm3 SDK is.

**Comment by garoto** [comment]
From this same thread by one of the dude directly involved in the mod:

> We will probably take a look at it after Episode 2 has beeen released.
> So the short answer is no.
> We don't have time atm.

How much more of a direct answer do you people need?

**Comment by DanielGibson** [comment]
You quoted the wrong post, Episode 2 has been released last September.

The relevant post is:
> Episode 2 took a lot longer than we anticipated. We're a small team and we're very focused on getting this game finished as soon as possible. So all our spare time will go towards the next episode before anything else.

Anyway, while it's great for me to see that people want to play Phobos with dhewm3, and I'm sure the Phobos developers also appreciate the interest for their Mod, you'll have to remain patient :)
Like dhewm3, Phobos is developed by people in their freetime, so they can only do so much and need to prioritize things.

**Comment by shmabler** [comment]
@SimonNJensen Could Phobos be made to work with Proton in the meantime?

**Comment by lmills87** [comment]
> You quoted the wrong post, Episode 2 has been released last September.
> 
> The relevant post is:
> 
> > Episode 2 took a lot longer than we anticipated. We're a small team and we're very focused on getting this game finished as soon as possible. So all our spare time will go towards the next episode before anything else.
> 
> Anyway, while it's great for me to see that people want to play Phobos with dhewm3, and I'm sure the Phobos developers also appreciate the interest for their Mod, you'll have to remain patient :) Like dhewm3, Phobos is developed by people in their freetime, so they can only do so much and need to prioritize things.

But, and if I'm understanding correctly, you've been offering to do the work for them, have you not?

It's not been 3 years since your last reply and still it seems no progress has been made here; count me among the others that would love to be able to enjoy Phobos on Dhewm3. A new episode just literally released; is now not a good time to let Daniel port the first 3?

Please forgive me if I'm speaking ignorantly, I just wanted to voice my support for this as I would love to see it happen.

**Comment by DanielGibson** [comment]
Yes, I'm still willing to port Phobos to dhewm3, I just need the source code :)

**Comment by walikor** [comment]
> Yes, I'm still willing to port Phobos to dhewm3, I just need the source code :)

![Cattura](https://github.com/user-attachments/assets/dd91f5a0-de6d-4c75-8ec2-8aca5fc04c65)


let's wait :)

**Comment by TwelveEyes** [comment]
> https://www.moddb.com/games/doom-iii/addons/doom-3-phobos-dhewm3-compatibility-patch

Compat patch until Team Future properly release their source code (if that ever happens).

**Comment by DanielGibson** [comment]
Nice!

**Comment by kingkriss** [comment]
> Yeah, for this to work the mods game dll sourcecode must be released under GPL and be ported to dhewm3 (see https://github.com/dhewm/dhewm3-sdk - I could do the porting if they give me the source).
> 
> I'm not in contact with the Phobos developers (yet?), but it could sure help if you (and everyone else who'd like to play it on dhewm3) told them about it so they know that people want this :-)

https://www.moddb.com/company/team-future
You probably know already. Regards.

**Comment by TwelveEyes** [comment]
> https://www.moddb.com/company/team-future You probably know already. Regards.

Team Future have refused to release their source code until completion of episode 4. Why? 🤷

In the meantime, you can use this:
https://github.com/dhewm/dhewm3/issues/213#issuecomment-2452740179

**Comment by SimonNJensen** [comment]
> > https://www.moddb.com/company/team-future You probably know already. Regards.
> 
> Team Future have refused to release their source code until completion of episode 4. Why? 🤷
> 
> In the meantime, you can use this: [#213 (comment)](https://github.com/dhewm/dhewm3/issues/213#issuecomment-2452740179)

Don't be deliberately obtuse.

We're not refusing anything. I've never said never. We've already stated several times why in this thread. 

We're getting pretty tired of having people telling us how to spend our time. Just because its important to you doesn't mean its important to us. We want to focus on making content with the time we have in between work and family.

**Comment by fgsfds** [comment]
No offence, but releasing source code is literally free. Releasing it would’ve taken less time than answering the same questions for years.

**Comment by SimonNJensen** [comment]
Dude, we know. As I’ve said before, it's not a priority for us and we're still making changes to the SDK for the next update

**Comment by IanHead** [comment]
Take your time. The episodes released so far are amazing, looking forward to the conclusion!

**Comment by REDHOTIRON2004** [comment]
> Dude, we know. As I’ve said before, it's not a priority for us and we're still making changes to the SDK for the next update

I just started playing Doom 3 source again after a long time through Batocera. Searched for the best recommended mods for Doom 3. And your Phobos mode came highly recommended by Google AI. Unfortunately for me, I couldn't get it working. On further research, I found a patch that didn't work either. Well, it is what it is.
I have kept this game on the playlist for the future, hopefully when it becomes fully compatible with dhewm3. Thanks for keeping this nostalgic game alive. And a future thanks from my future self, whenever I might be able to play this. 👍
